src/plugins/mer/merbuildengineoptionswidget.cpp

```diff
@@ -609,7 +609,25 @@ void MerBuildEngineOptionsWidget::update()
 
     if (show) {
         const auto sshParameters = buildEngine->virtualMachine()->sshParameters();
+
+        const bool wasBlocked = m_ui->buildEngineDetailsWidget->blockSignals(true); // Fix jb49638
```

Comment:
Remove lines 613 and 629, then asserts in lines 624 and 626 will trigger in case of apply for Memory and Processor(s) changes.